### ISSUE TITLE
iced_glow 0.5.1 crashing with 'location' : not supported for this version or the enabled extensions (GL v3.2)

### ISSUE THREAD

**Issue by greatest-ape** [iced-rs/iced#1625]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Is this issue related to iced?

- [X] My hardware is compatible and my graphics drivers are up-to-date.

### What happened?

Multiple [OctaSine](https://github.com/greatest-ape/OctaSine) users are seeing crashes since updating to iced_glow 0.5. [Detailed issue with error logs](https://github.com/greatest-ape/OctaSine/issues/135). OctaSine currently uses GL v3.2.

I think the crash is due to [using location here](https://github.com/iced-rs/iced/blame/a6d0d5773f0561a841a84b538523cbd97e91eccd/glow/src/shader/common/gradient.frag#L10) without enabling [ARB_explicit_attrib_location](https://registry.khronos.org/OpenGL/extensions/ARB/ARB_explicit_attrib_location.txt) when [loading the shader](https://github.com/iced-rs/iced/blob/master/glow/src/program.rs). This seems to be a very similar issue to https://github.com/iced-rs/iced/issues/1336, which was resolved by [these changes to glow_glyph](https://github.com/hecrj/glow_glyph/pull/7/files).

### What is the expected behavior?

No crash on supported GL versions.

### Version

0.6

### Operative System

Windows

### Do you have any log output?

```shell
20:15:01 [ERROR] thread '<unnamed>' panicked at 'ERROR: 0:13: 'location' : not supported for this version or the enabled extensions 
ERROR: 0:13: '' : missing #endif 
ERROR: 0:13: '' : compilation terminated 
ERROR: 3 compilation errors.  No code generated.
```

**Comment by greatest-ape** [comment]
Nice! I've made a build with your branch active and asked the original issue submitters to try it out.

**Comment by greatest-ape** [comment]
One of them has answered and says that the crash indeed gets fixed!

Does your PR contain breaking changes? I’m thinking about the chances of getting it released relatively soon :-)

**Comment by greatest-ape** [comment]
Ok :-)

**Comment by greatest-ape** [comment]
I've created a smaller PR for fixing this issue: https://github.com/iced-rs/iced/pull/1703

**Comment by greatest-ape** [comment]
Fixed through #1703
